provider/azure/terraform/example/helpers.rb

```diff
@@ -11,13 +11,37 @@ def get_example_properties_to_check(example_name, object)
             params = param_props.map{|p| p.name.underscore}.to_set
 
             example = get_example_by_names(example_name)
-            example.properties
+            example_props = example.properties
               .reject do |pn, pv|
                 params.include?(pn) || pn == 'location'
               end
               .transform_values do |v|
                 v.is_a?(String) && !v.match(/\$\{.+\}/).nil? ? :AttrSet : v
               end
+            flatten_example_properties_to_check(example_props, true)
+          end
+
+          def flatten_example_properties_to_check(properties, is_nested)
+            return properties unless is_nested
+            flat_properties = Hash.new
+            next_is_nested = false
+            properties.each do |pn, pv|
+              if pv.is_a?(Hash)
+                pv.each do |key, val|
+                  flat_properties[pn+"."+key] = val
```

Comment:
You can simplify the string with `"#{pn}.#{key}"`.

Reply:
Replaced! Also done for other similar cases.